mobius-core/src/test/java/com/spotify/mobius/MobiusLoopDisposalBehavior.java

```diff
@@ -276,6 +280,50 @@ public void shouldDisposeMultiThreadedEventSourceSafely() throws Exception {
     }
   }
 
+  @Test
+  public void shouldSafelyDisposeWhenDisposeAndEventsAreOnDifferentThreads() throws Exception {
+    final Random random = new Random();
+    final MobiusLoop.Builder<String, TestEvent, TestEffect> builder =
+        Mobius.loop(update, effectHandler)
+            .eventRunner(
+                new Producer<>() {
+                  @Nonnull
+                  @Override
+                  public WorkRunner get() {
+                    return WorkRunners.from(Executors.newFixedThreadPool(4));
+                  }
+                });
+    new Thread(
```

Comment:
You should ideally keep this thread in a variable, and `join` with it at the end of the test, to ensure that it finishes and that any exceptions thrown are reported.